From 6b178b0eb5c28200faad5273780a5227a2ed7d7a Mon Sep 17 00:00:00 2001
From: Juan Pablo Novillo <juampy72@gmail.com>
Date: Wed, 11 Apr 2012 02:47:26 +0200
Subject: [PATCH] Issue #1371700 by juampy, merlinofchaos, prdctvtxt and tim.plunkett:
 CTools adds cached plugins with wrong paths during hook_registry_files_alter() in multisite systems

Reverted changes as this is a core bug. See http://drupal.org/node/1372122#comment-5854040
---
 includes/plugins.inc |   15 ++++-----------
 1 files changed, 4 insertions(+), 11 deletions(-)

diff --git a/includes/plugins.inc b/includes/plugins.inc
index a6d4bd6..2dbdd48 100644
--- a/includes/plugins.inc
+++ b/includes/plugins.inc
@@ -246,17 +246,10 @@ function ctools_get_plugins($module, $type, $id = NULL) {
     $cache = cache_get("plugins:$module:$type", $info[$module][$type]['cache table']);
 
     if (!empty($cache->data)) {
-      // Verify that the file exists as it may have been moved in a multisite environment
-      if (isset($cache->data['path'])) {
-        // Files to be registered may be in two locations. Check for both of them.
-        if ((is_array($cache->data['path']) && file_exists($cache->data['path']['file'])) ||
-           file_exists($cache->data['path'])) {
-          // Cache load succeeded so use the cached plugin list.
-          $plugins[$module][$type]   = $cache->data;
-          // Set $setup to true so we know things where loaded.
-          $setup[$module][$type]     = TRUE;
-        }
-      }
+      // Cache load succeeded so use the cached plugin list.
+      $plugins[$module][$type]   = $cache->data;
+      // Set $setup to true so we know things where loaded.
+      $setup[$module][$type]     = TRUE;
     }
     else {
       // Cache load failed so store that we need to build and write the cache.
-- 
1.7.4.1

